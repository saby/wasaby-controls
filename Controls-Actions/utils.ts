import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    documentContext?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
