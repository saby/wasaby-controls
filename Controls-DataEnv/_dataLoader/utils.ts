import { loadAsync } from 'WasabyLoader/ModulesLoader';
import { addPageDeps } from 'UI/Deps';
import { logger as Logger } from 'Application/Env';
import { RPC } from 'Browser/Transport';

function resolveLogger(error: Error): typeof Logger.error | typeof Logger.warn {
    let logger = Logger.error;

    if (error instanceof RPC.Error) {
        logger = error.errType === 'warning' ? Logger.warn : Logger.error;
    }

    return logger;
}

export async function loadModule<TModule>(moduleName: string): Promise<TModule | undefined> {
    try {
        const module = await loadAsync<TModule>(moduleName);
        addPageDeps([moduleName]);
        return module;
    } catch (error: unknown) {
        loadErrorHandler(
            error,
            `Controls-DataEnv/dataLoader::Ошибка при загрузке модуля ${moduleName}`
        );
    }
}

export function loadErrorHandler(e: unknown, message: string): void {
    if (e instanceof Error) {
        const logger = resolveLogger(e);
        logger(`${message} ${e.message}`);
    }
}
