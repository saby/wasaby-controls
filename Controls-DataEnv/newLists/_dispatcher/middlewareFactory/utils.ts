export const addName = <T extends Function>(label: string, fn: T): T => {
    const wrapper = {
        [label]: ((...args: unknown[]) => fn(...args)) as unknown as T,
    };

    return wrapper[label];
};

const POSTFIX = '{name}';
export const createMask = <T extends string>(prefix: T) => `${prefix}(${POSTFIX})` as const;
export const decorateName = (mask: string, name: string) => mask.replace(POSTFIX, name);
export const undecorateName = (mask: string, decoratedName: string) => {
    const [start, end] = mask.split(POSTFIX);
    return decoratedName.replace(start, '').replace(end, '');
};
