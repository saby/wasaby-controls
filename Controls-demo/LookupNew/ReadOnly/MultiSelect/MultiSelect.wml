<div class="controlsDemo__wrapper controlsDemo__flexColumn">
    <div class="controlsDemo__mb1 controlsDemo__wrapper__horizontal controlsDemo__cell controlsDemo__childWidth400 controlsDemo-Lookup-simple">
        <Controls.lookup:Input
                bind:selectedKeys="_selectedKeys"
                source="{{_source}}"
                searchParam="title"
                keyProperty="id"
                multiSelect="{{true}}"
                readOnly="{{true}}">
            <ws:itemTemplate>
                <ws:partial template="Controls/lookup:ItemTemplate"
                            displayProperty="title"/>
            </ws:itemTemplate>
        </Controls.lookup:Input>
    </div>
</div>