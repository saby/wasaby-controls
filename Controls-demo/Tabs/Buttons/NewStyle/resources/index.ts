import { RecordSet } from 'Types/collection';

export const ITEMS = new RecordSet({
    keyProperty: 'id',
    rawData: [
        {
            id: '1',
            title: 'Document',
        },
        {
            id: '2',
            title: 'Files',
        },
        {
            id: '3',
            title: 'Orders',
        },
    ],
});

export const ITEMS_WITH_ICON_AND_COUNTER = new RecordSet({
    keyProperty: 'id',
    rawData: [
        {
            id: '1',
            title: 'Document',
        },
        {
            id: '2',
            title: 'Files',
            mainCounter: 3,
        },
        {
            id: '3',
            title: 'Orders',
            icon: 'icon-EmptyMessage',
            iconStyle: 'label',
        },
    ],
});
