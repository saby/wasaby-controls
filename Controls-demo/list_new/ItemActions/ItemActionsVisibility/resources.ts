import { IoC } from 'Env/Env';
import { Model } from 'Types/entity';

import { IItemAction, TItemActionShowType } from 'Controls/itemActions';

export interface ISrcData {
    key: number;
    title: string;
    description: string;
}

export const columns = [
    {
        displayProperty: 'key',
    },
    {
        displayProperty: 'title',
    },
    {
        displayProperty: 'description',
    },
];

export const srcData: ISrcData[] = [
    {
        key: 1,
        title: 'Notebooks 1',
        description: 'Другое название 1',
    },
    {
        key: 2,
        title: 'Notebooks 2',
        description: 'Описание вот такое',
    },
    {
        key: 3,
        title: 'Smartphones 3 ',
        description: 'Хватит страдать',
    },
    {
        key: 4,
        title: 'Notebooks 1',
        description: 'Другое название 1',
    },
    {
        key: 5,
        title: 'Notebooks 2',
        description: 'Описание вот такое',
    },
    {
        key: 6,
        title: 'Smartphones 3 ',
        description: 'Хватит страдать',
    },
];

export const menuItemActions: IItemAction[] = [
    {
        id: 1,
        icon: 'icon-PhoneNull',
        title: 'phone',
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action phone Click ', model);
        },
    },
    {
        id: 2,
        icon: 'icon-EmptyMessage',
        title: 'message',
        parent: null,
        'parent@': true,
        handler(model: Model): void {
            alert('Message Click');
        },
    },
    {
        id: 3,
        icon: 'icon-Profile',
        title: 'profile',
        showType: TItemActionShowType.MENU,
        parent: 2,
        'parent@': null,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action profile Click');
        },
    },
    {
        id: 6,
        title: 'call',
        parent: 2,
        'parent@': null,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action profile Click');
        },
    },
];

export const menuToolbarItemActions: IItemAction[] = [
    {
        id: 5,
        title: 'прочитано',
        showType: TItemActionShowType.TOOLBAR,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action read Click');
        },
    },
    ...menuItemActions,
    {
        id: 4,
        icon: 'icon-Erase',
        iconStyle: 'danger',
        title: 'delete pls',
        showType: TItemActionShowType.TOOLBAR,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action delete Click');
        },
    },
];
