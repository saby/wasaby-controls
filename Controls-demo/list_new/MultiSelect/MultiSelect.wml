<div class="controlsDemo__wrapper">
    <div class="controlsDemo__cell controlsDemo__mr4">
        <Router.router:Reference attr:class="controls-text-label"
                                 state="/Controls-demo/app/:app"
                                 app="{{ 'Controls-demo/list_new/MultiSelect/MultiSelectVisibility/Index' }}">
            <a href="{{content.href}}">Режим отображения чекбоксов</a>
        </Router.router:Reference>
        <Controls-demo.list_new.MultiSelect.MultiSelectVisibility.Index/>
    </div>

    <div class="controlsDemo__cell controlsDemo__mr4">
        <Router.router:Reference attr:class="controls-text-label"
                                 state="/Controls-demo/app/:app"
                                 app="{{ 'Controls-demo/list_new/MultiSelect/AllSelected/Index' }}">
            <a href="{{content.href}}">Выбор сразу всех записей</a>
        </Router.router:Reference>
        <Controls-demo.list_new.MultiSelect.AllSelected.Index/>
    </div>

    <div class="controlsDemo__cell controlsDemo__mr4">
        <Router.router:Reference attr:class="controls-text-label"
                                 state="/Controls-demo/app/:app"
                                 app="{{ 'Controls-demo/list_new/MultiSelect/CustomPosition/Index' }}">
            <a href="{{content.href}}">Нестандартное расположение чекбоксов</a>
        </Router.router:Reference>
        <Controls-demo.list_new.MultiSelect.CustomPosition.Index/>
    </div>
</div>
