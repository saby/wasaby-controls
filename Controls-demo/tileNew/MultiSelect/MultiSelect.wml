<div class="controlsDemo__wrapper controlDemo_Tile_MultiSelect">
<Controls.treeTile:View
        tileMode="dynamic"
        displayProperty="title"
        imageProperty="image"
        keyProperty="id"
        multiSelectVisibility="visible"
        bind:selectedKeys="_selectedKeys"
        bind:excludedKeys="_excludedKeys"
        source="{{_viewSource}}">
        <ws:itemTemplate>
                <ws:partial
                        template="Controls/tile:ItemTemplate"
                        hasTitle="{{itemTemplate.item.contents.isDocument}}"/>
        </ws:itemTemplate>
</Controls.treeTile:View>
</div>
