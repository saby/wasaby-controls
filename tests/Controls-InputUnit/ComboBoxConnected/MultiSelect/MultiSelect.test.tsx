/**
 * @jest-environment jsdom
 */

import { getContent } from 'Controls-InputUnit/resources/utils';
import { WasabyEvents } from 'UICore/Events';
import { unmountComponentAtNode } from 'react-dom';
import { render } from '@testing-library/react';
import { default as MultiSelect } from 'Controls-Input-demo/ComboboxConnected/MultiSelect';

describe('Controls-Input/ComboBoxGroupConnected/MultiSelect', () => {
    setTestID(152174963);
    let container = null;

    beforeEach(() => {
        jest.useFakeTimers('modern');
        jest.setSystemTime(new Date(1996, 5, 21));
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });
    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        jest.useRealTimers();
        container.remove();
        container = null;
    });

    it('MultiSelect', async () => {
        render(getContent(MultiSelect), { container });
        expect(container).toMatchSnapshot();
    });
});
