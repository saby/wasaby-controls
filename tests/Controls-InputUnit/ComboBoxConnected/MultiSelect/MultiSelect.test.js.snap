// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/ComboBoxGroupConnected/MultiSelect MultiSelect 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth300 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            Multiselect = true
          </div>
          
        </div>
      </a>
      <div
        class="controls-Dropdown controls-margin_top-m"
        tabindex="0"
      >
        <div
          class="ws-inline-flexbox controls-Dropdown__infoboxWrapper"
        >
          <div
            class="controls-Dropdown__wrapper controls-Dropdown-fontsize controls-Dropdown__wrapper-height-default "
          >
            <div
              class="controls-Dropdown__content-wrapper controls-Dropdown__content-wrapper-height-default  controls-Dropdown__text_enabled controls-Dropdown__item_style-undefined controls-Dropdown__defaultItem controls-text-link"
              name="popupTarget"
              tabindex="0"
            >
              <div
                class="controls-Dropdown__text-wrapper controls-Dropdown__text-wrapper-hovered"
                title="Первый"
              >
                <div
                  class="controls-Dropdown__text"
                  data-qa="controls-Dropdown__content"
                >
                  Первый
                </div>
                <span
                  class="controls-Dropdown__hasMoreText"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            Multiselect = false
          </div>
          
        </div>
      </a>
      <div
        class="controls-ComboBox controls_toggle_theme-undefined controls-ComboBox-borderStyle-valid   controls-ComboBox-borderStyle-valid_focusing controls-ComboBox_background-same controls-text-default controls-fontweight-default controls-inlineheight-default controls-Combobox-fontsize-m controls-ComboBox_textAlign-left controls-ComboBox_border-partial controls-margin_top-m"
        tabindex="0"
        title=""
      >
        <div
          class="controls-ComboBox-wrapper"
        >
          <div
            class="controls-ComboBox__field controls-ComboBox__field_margin-null "
            tabindex="0"
          >
            <div
              class="controls-ComboBox__placeholder  tw-cursor-text"
            >
              Выберите вариант
            </div>
          </div>
          <span
            class="controls-ComboBox__arrow-wrapper controls-ComboBox__arrow-wrapper_offset-null"
          >
            <span
              class="controls-ComboBox__iconArrow icon-ExpandLight"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
